src/script/lua_api/l_util.cpp

```diff
@@ -272,11 +272,11 @@ int ModApiUtil::l_compress(lua_State *L)
 	const char *data = luaL_checklstring(L, 1, &size);
 
 	int level = -1;
-	if (!lua_isnone(L, 3) && !lua_isnil(L, 3))
-		level = readParam<float>(L, 3);
+	if (!lua_isnoneornil(L, 3))
+		level = readParam<int>(L, 3);
 
-	std::ostringstream os;
-	compressZlib(std::string(data, size), os, level);
+	std::ostringstream os(std::ios_base::binary);
+	compressZlib(reinterpret_cast<const u8*>(data), size, os, level);
```

Comment:
```suggestion
	compressZlib(reinterpret_cast<const u8 *>(data), size, os, level);
```
^ Pedanticism.

(Btw., I find it awful how the 2nd argument is just ignored here, to silently ignore mod errors. But that's out of scope.)